droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/ProfileContextLocator.java

```diff
@@ -322,6 +328,30 @@ private void generateNewDatabaseAndTemplates(final ProfileInstance profile,
             }
         } 
     }
+
+    private TemplateStatus checkModificationOfBinarySignature(String signatureFileName, TemplateStatus status) {
+
+        try {
+            Path binarySignaturePath = globalConfig.getSignatureFileDir().resolve(signatureFileName);
+            if (binarySignaturePath.toFile().exists()) {
+                String binarySignatureLastUpdatedName = BINARY_SIGNATURE_LAST_UPDATED.getName() + "." + URLEncoder.encode(signatureFileName, Charset.defaultCharset());
+                long lastModifiedTime = Files.getLastModifiedTime(binarySignaturePath).toMillis();
+                if (!globalConfig.getProperties().containsKey(binarySignatureLastUpdatedName)) {
+                    globalConfig.update(Map.of(binarySignatureLastUpdatedName, lastModifiedTime));
+                    return status;
+                } else {
+                    long lastModifiedFromConfig = globalConfig.getProperties().getLong(binarySignatureLastUpdatedName);
+                    if (lastModifiedTime > lastModifiedFromConfig) {
+                        globalConfig.update(Map.of(binarySignatureLastUpdatedName, lastModifiedTime));
+                        return TemplateStatus.NO_TEMPLATE;
+                    }
+                }
+            }
+            return status;
+        } catch (ConfigurationException | IOException e) {
+            throw new RuntimeException("Error checking the last modified status of the binary signature file", e);
```

Comment:
I know this will be rare situation but, if we can't detect the last modified time, should we stop and throw exception? Or just log this situation as warning and carry on like how we have it currently?

Reply:
I switched back and forth as to whether this should be a warning or an error and I think you're right, it's a bit risky throwing an error here considering most users won't ever modify the signature file. I've changed it to a `log.warn` call